Mark config-trusted-resources as deprecated

Yongxuanzhang opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

This commit marks config-trusted-resources as deprecated, VerificationPolicy has covered the features of config-trusted-resources so there's no need to keep it.

/kind misc

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

<!-- Describe your changes here- ideally you can get that description straight from your descriptive commit message(s)! -->

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: config-trusted-resources of trusted resources to store public key is deprecated and will be removed in release 0.46, please use VerificationPolicy instead 
```

JeromeJu commented on pkg/apis/config/trusted_resources.go:
```diff
@@ -29,6 +29,7 @@ import (
 // TrustedResources holds the collection of configurations that we attach to contexts.
 // Configmap named with "config-trusted-resources" where cosign pub key path and
 // KMS pub key path can be configured
+// Deprecated.
```
Maybe we would like to add 'Use verification policy instead' for pointing users to the alternative.

Yongxuanzhang replied:
Sure! Thanks for the suggestion!

JeromeJu reviewed (approved):
Thanks Yongxuan

JeromeJu commented:
Maybe we would like to add 'Use verification policy instead' for pointing users to the alternative.
^ I think this might also apply to the release note

lbernick commented on docs/deprecations.md:
```diff
@@ -26,7 +26,7 @@ being deprecated.
 | [ClusterTask is deprecated](https://github.com/tektoncd/pipeline/issues/4476)                                                                                                                                                | v0.41.0                                                              | Beta                                                                                                    ...
 | [`pipelineRef.bundle` and `taskRef.bundle` are deprecated](https://github.com/tektoncd/pipeline/issues/5514)                                                                                                                 | v0.41.0                                                              | Alpha                                                                                                   ...
-
+| [`config-trusted-resources` is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | April 10, 2023 ...
```
Do we need to update the alpha features table to explain that this feature is going to be removed? I'm also confused what the "config-trusted-resources" feature actually is. Is that the feature enabled by the flag "resource-verification-mode"? How would someone use VerificationPolicy, if that's the replacement feature?
Lastly, why are we deciding to support this for two more months?

Yongxuanzhang replied:
- This feature stores public keys in config map, it's a subset feature of VerificationPolicy. 
- I will update the docs as @JeromeJu suggested. 
- The announcement is 0.45, are we ok to remove it in 0.46? If so I think we could change it to 0.46

lbernick replied:
sg, please lmk when you've updated the PR to make it more clear how to use the replacement feature. re: when can we remove it? we can remove alpha features at any time according to our compatibility policy, so I'd put N/A (alpha)

lbernick commented on pkg/apis/config/trusted_resources.go:
```diff
@@ -29,6 +29,7 @@ import (
 // TrustedResources holds the collection of configurations that we attach to contexts.
 // Configmap named with "config-trusted-resources" where cosign pub key path and
 // KMS pub key path can be configured
+// Deprecated.
```
I'm surprised you don't need to re-run apigen here, what's going on?

Yongxuanzhang replied:
I think this is config not api in v1beta1 or v1? when I run `update-openapigen.sh` nothing happens

lbernick replied:
Ah I see, we only generate deepcopy functions, thanks!

lbernick commented on docs/deprecations.md:
```diff
@@ -26,7 +26,7 @@ being deprecated.
 | [ClusterTask is deprecated](https://github.com/tektoncd/pipeline/issues/4476)                                                                                                                                                | v0.41.0                                                              | Beta                                                                                                    ...
 | [`pipelineRef.bundle` and `taskRef.bundle` are deprecated](https://github.com/tektoncd/pipeline/issues/5514)                                                                                                                 | v0.41.0                                                              | Alpha                                                                                                   ...
-
+| [`config-trusted-resources` is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | v0.46.0        ...
```
```suggestion
| [The `config-trusted-resources` configMap is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | v0.46.0                       |
```

pritidesai commented:
please add a little more details in the release note, I do no think so but could anyone be using this configmap? Also, how about we also update the doc at https://github.com/tektoncd/pipeline/blob/main/docs/trusted-resources.md#config-key-at-configmap-will-be-deprecated?

Yongxuanzhang commented:
> please add a little more details in the release note, I do no think so but could anyone be using this configmap? Also, how about we also update the doc at https://github.com/tektoncd/pipeline/blob/main/docs/trusted-resources.md#config-key-at-configmap-will-be-deprecated?

Thanks! I have updated the release notes. For the docs section, I think I have updated? What else am I missing?

afrittoli commented:
/test pull-tekton-pipeline-go-coverage-df

XinruZhang commented:
It would be great if we can create an issue to track the removal of this ConfigMap :)
Thank you @Yongxuanzhang

abayer commented:
/lgtm
